ush/metplus/decouple_fcst_obs_vx_config.py

```diff
@@ -0,0 +1,456 @@
+import pprint
+import subprocess
+
+from pathlib import Path
+file = Path(__file__).resolve()
+home_dir = file.parents[2]
+ush_dir = Path(os.path.join(home_dir, 'ush')).resolve()
+sys.path.append(str(ush_dir))
+
+from python_utils import (
+    log_info,
+    load_config_file,
+)
+
+
+def get_pprint_str(var, indent_str=''):
+    """
+    Function to format a python variable as a pretty-printed string and add
+    indentation.
+
+    Arguments:
+    ---------
+    var:
+      A variable.
+
+    indent_str:
+      String to be added to the beginning of each line of the pretty-printed
+      form of var.  This usually consists of multiple space characters.
+
+    Returns:
+    -------
+    var_str:
+      Formatted string containing contents of variable.
+    """
+
+    var_str = pprint.pformat(var, compact=True, sort_dicts=False)
+    var_str = var_str.splitlines(True)
+    var_str = [indent_str + s for s in var_str]
+    var_str = ''.join(var_str)
+
+    return var_str
+
+
+def create_pprinted_msg(vars_dict, indent_str='', add_nl_after_varname=False):
+    """
+    Function to create an output message (string) containing one or more
+    variables' names, with each name followed possibly by a newline, an equal
+    sign, and the pretty-printed value of the variable.  Each variable name
+    starts on a new line.
+
+    Arguments:
+    ---------
+    vars_dict:
+      Dictionary containing the variable names (the keys) and their values
+      (the values).
+
+    indent_str:
+      String to be added to the beginning of each line of the string before
+      returning it.  This usually consists of multiple space characters.
+
+    add_nl_after_varname:
+      Flag indicating whether to add a newline after the variable name (and
+      before the equal sign).
+
+    Returns:
+    -------
+    vars_str:
+      Formatted string containing contents of variable.
+    """
+
+    space_or_nl = ' '
+    one_or_zero = 1
+    if add_nl_after_varname:
+        space_or_nl = '\n'
+        one_or_zero = 0
+
+    vars_str = ''
+    for var_name, var_value in vars_dict.items():
+        pprint_indent_str = ' '*(2 + one_or_zero*(1 + len(var_name)))
+        tmp = f'{var_name}' + space_or_nl + '= ' + \
+            get_pprint_str(var_value, pprint_indent_str).lstrip()
+        vars_str = '\n'.join([vars_str, tmp])
+
+    vars_str = indent(vars_str, indent_str)
+
+    return vars_str
+
+
+def extract_fcst_obs_vals_from_cpld(item_cpld):
+    """
+    Function to parse the "coupled" value of an item (obtained from the coupled
+    verification (vx) configuration dictionary) to extract from it the item's
+    value for forecasts and its value for observations.  The coupled item
+    (item_cpld) is a string that may correspond to a field name, a level, or
+    a threshold.  It can be the name of a key in a key-value pair in the
+    coupled vx configuration dictionary, or it can be the value (or, if the
+    value is a list of strings, one of the elements in that list).
```

Comment:
Suggest cutting down less relevant information to make the docstring more readable

```suggestion
    Function to parse the "coupled" value of an item (obtained from the coupled
    verification (vx) configuration dictionary) to extract from it the item's
    value for forecasts and its value for observations.  The coupled item
    (item_cpld) is a string that may correspond to a field name, a level, or
    a threshold.
```

Reply:
Shortened.  Will show up in the next commit(s) I push.